menu.go

```diff
@@ -3,24 +3,28 @@ package fyne
 // Menu stores the information required for a standard menu.
 // A menu can pop down from a MainMenu or could be a pop out menu.
 type Menu struct {
-	Label string
-	Items []*MenuItem
+	Label            string
+	Items            []*MenuItem
+	AfterNativeMenus bool
 }
 
 // NewMenu creates a new menu given the specified label (to show in a MainMenu) and list of items to display.
 func NewMenu(label string, items ...*MenuItem) *Menu {
-	return &Menu{label, items}
+	return &Menu{Label: label, Items: items}
 }
 
-// MenuItem is a sligne item within any menu, it contains a dispay Label and Action function that is called when tapped.
+// MenuItem is a single item within any menu, it contains a display Label and Action function that is called when tapped.
 type MenuItem struct {
-	Label  string
-	Action func()
+	Label             string
+	KeyEquivalent     string
+	PlaceInNativeMenu bool
```

Comment:
"NativeMenu" seems like a term that may not be familiar to people.
How often will it be used? I don't think that mac guidelines support putting lots of stuff in this category, though I can't find the doc at this time.
Isn't it just the "About", "Quit" and the "Preferences" items? In which case perhaps there is a cleaner way where these are special types, so "placeInNative" becomes an internal detail when running on mac in native menu mode?

Reply:
I don't think that there is a standard set of app menu entries beside the system entries (About, Services, Hide, Hide Others, Show All, Quit) and Settings. I checked five different apps and they all have different app menu entries and some even have different app menu layout.
I think that we are too inflexible if we restrict the set of menu items that can be put into the native app menu. If a developer doesn't care for menu item layout on MacOS, he simply creates his menus with the default constructor and doesn't think about native menus.

Reply:
> If a developer doesn't care for menu item layout on MacOS, he simply creates his menus with the default constructor and doesn't think about native menus.

I disagree with this. We should not expect developers to “care” about macOS, we should do they for them. For example the “Quit” item is automatically in the correct place. If the developer wants a “Preferences” item in their menu then on macOS it should be in the app-menu as that is what the OS expects. As in other places I’d rather stick to the semantics rather than direct configuration. “This is a quit item” vs “this goes at the bottom of native menu if on macOS”.

Reply:
I agree with you in this two cases and probably in some others (check for updates, help and about). However, a lot of apps place additional app-specific entries in the app menu. Your approach does not allow this.

Reply:
Very true. Is it worth looking at the specific cases and trying to figure out an approach when we know more?